### PR TITLE
feat(e2e): nv/pob integration init [ENG-1507]

### PR DESCRIPTION
**Pull request by nivasan1**
## In This PR
 - I scaffold the necessary helpers / setup code to migrate our current e2e testing suite into its own submodule / utilize interchain test

### CODE REVIEW

File: tests/integration/chain_setup.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	interchaintest "github.com/strangelove-ventures/interchaintest/v7"
+	"github.com/strangelove-ventures/interchaintest/v7/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v7/ibc"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+// ChainBuilderFromChainSpec creates an interchaintest chain builder factory given a ChainSpec
+// and returns the associated chain
+func ChainBuilderFromChainSpec(t *testing.T, spec *interchaintest.ChainSpec) ibc.Chain {
```
Comment (davidterpay) [inline tests/integration/chain_setup.go:19]: nit: I would make all of these inherit the the test suite. Why are you passing `testing.T` directly?

File: tests/integration/chain_setup.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	// require that NumFullNodes == NumValidators == 4
+	require.Equal(t, spec.NumFullNodes, spec.NumValidators)
```
Comment (davidterpay) [inline tests/integration/chain_setup.go:21]: nit: Do we want to make this a global variable?

File: tests/integration/chain_setup.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+func CreateAuctionBidMsg(t *testing.T, ctx context.Context, searcher cosmos.User, chain *cosmos.CosmosChain, bid sdk.Coin, users []MessagesForUser) *buildertypes.MsgAuctionBid {
+	// for each MessagesForUser get the signed bytes
+	txs := make([][]byte, len(users))
+	for i, user := range users {
+		txs[i] = CreateTx(t, ctx, chain, user.User, user.Msgs...)
+	}
+
+	bech32SearcherAddress := searcher.FormattedAddress()
+	accAddr, err := sdk.AccAddressFromBech32(bech32SearcherAddress)
+	require.NoError(t, err)
```
Comment (davidterpay) [inline tests/integration/chain_setup.go:101]: nit: i personally find it easier to work with `suite.Suite` bc you can just do `s.Require().Condition...` but if this is convention then im down.

**Review by davidterpay — COMMENTED** [review]
left some comments

File: tests/integration/chain_setup.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	// create a message auction bid
+	return buildertypes.NewMsgAuctionBid(
+		accAddr,
+		bid,
+		txs,
+	)
+}
+
+// BroadcastMsg broadcasts the given messages as a tx signed by the given sender, it blocks until a response from the chain is received
+// and fails if a timeout occurs
```
[the remark this answers is not in the document]
Comment (nivasan1) [reply tests/integration/chain_setup.go:112]: Tx inclusion

**Review by davidterpay — APPROVED** [review]

**Pull request closed without merging**